### PR TITLE
feat: add --open flag to open browser on successful start of the server

### LINKED ISSUE #503: Open the browser on successful start
Open the browser with the URL and the PORT on the successful start of the server.

### PR DESCRIPTION
**Pull request by IdoKendo**
**Description**

This PR fixes #503 

This PR adds the optional `--open` flag to use as a startup argument, which, when added, will launch the browser with the URL and PORT upon the successful start of the server.

Example usage:
```
python integration_tests/base_routes.py  --open
```
Will open the browser:
image

### CODE REVIEW

**Comment by sansyrox** [comment]
Thank you for the PR @IdoKendo 😄 

I have a few comments here. But before that -- are you on our discord?

**Comment by IdoKendo** [comment]
> Thank you for the PR @IdoKendo 😄
> 
> I have a few comments here. But before that -- are you on our discord?

No, I am not on Discord at all, but if it's a more convenient way I can join. 😃

**Comment by sansyrox** [comment]
@IdoKendo , that will be great 😊 
It will be easier to communicate there 😅

https://discord.gg/mX5cGYfx

**Comment by IdoKendo** [comment]
> @IdoKendo , that will be great 😊 It will be easier to communicate there 😅
> 
> https://discord.gg/mX5cGYfx

Joined! 😄

**Comment by sansyrox** [comment]
Hey @IdoKendo 👋 

Great work with the PR and apologies for this late review. I have a suggestion for this PR.


Can we parse an environment variable like `ROBYN_BROWSER_OPEN=TRUE` from the `robyn.env` file? And set the default value to true here? And change the flag from `--open` to `--open-browser` ?

Let me know what you think of it.

**Comment by IdoKendo** [comment]
> Hey @IdoKendo 👋
> 
> Great work with the PR and apologies for this late review. I have a suggestion for this PR.
> 
> Can we parse an environment variable like `ROBYN_BROWSER_OPEN=TRUE` from the `robyn.env` file? And set the default value to true here? And change the flag from `--open` to `--open-browser` ?
> 
> Let me know what you think of it.

Yeah, no problem!

Are you sure about setting the default to true though? In my opinion, opening the browser on launching the server should be an opt-in decision. WDYT?

**Comment by sansyrox** [comment]
> Are you sure about setting the default to true though? In my opinion, opening the browser on launching the server should be an opt-in decision. WDYT?

Yes, you're right. I agree with it. But we should add some documentation about it. I was worried that most people won't use it if it was not explicit enough.

**Comment by IdoKendo** [comment]
> > Are you sure about setting the default to true though? In my opinion, opening the browser on launching the server should be an opt-in decision. WDYT?
> 
> Yes, you're right. I agree with it. But we should add some documentation about it. I was worried that most people won't use it if it was not explicit enough.

Yes, I agree with this notion. I can add some documentation about it, maybe even in the README file, WDYT?

**Comment by sansyrox** [comment]
> > > Are you sure about setting the default to true though? In my opinion, opening the browser on launching the server should be an opt-in decision. WDYT?
> > 
> > 
> > Yes, you're right. I agree with it. But we should add some documentation about it. I was worried that most people won't use it if it was not explicit enough.
> 
> Yes, I agree with this notion. I can add some documentation about it, maybe even in the README file, WDYT?

Adding it to the README is an interesting idea. How would you do that??

**Comment by IdoKendo** [comment]
> > > > Are you sure about setting the default to true though? In my opinion, opening the browser on launching the server should be an opt-in decision. WDYT?
> > > 
> > > 
> > > Yes, you're right. I agree with it. But we should add some documentation about it. I was worried that most people won't use it if it was not explicit enough.
> > 
> > 
> > Yes, I agree with this notion. I can add some documentation about it, maybe even in the README file, WDYT?
> 
> Adding it to the README is an interesting idea. How would you do that??

Maybe under the usage/run your code section?

**Comment by sansyrox** [comment]
> Maybe under the usage/run your code section?

Okay, sure. We can give this a go :D

**Comment by IdoKendo** [comment]
> > Maybe under the usage/run your code section?
> 
> Okay, sure. We can give this a go :D

Added, LMK if there's anything else to approve this PR 😄

File: README.md [hunk 1]
```diff
@@ -64,8 +64,16 @@ options:
   --workers WORKERS         Choose the number of workers. [Default: 1]
   --dev                     Development mode. It restarts the server based on file changes.
   --log-level LOG_LEVEL     Set the log level name
+  --create                  Create a new project template.
+  --docs                    Open the Robyn documentation.
+  --open-browser            Open the browser on successful start.
```
Comment (sansyrox) [inline README.md:69]: Ooo. Love this. Thank you! :D

**Review by sansyrox — APPROVED** [review]
Looks great! 🔥

**Pull request merged**